### DISCUSSION TITLE
Unable to run meteor

### DISCUSSION THREAD

**Discussion by mynameismon** [edemaine/coauthor#590] [Installation] [answered]
## What's wrong?
On running the steps reffered to in the [installation guide](https://github.com/edemaine/coauthor/blob/main/INSTALL.md), and starting the app by running ``meteor``, I am greeted with a massive wall of text

## How to Reproduce
1. Clone the repository into a fresh VM
2. Run ``meteor npm install``
3. Run ``meteor``

## Logs:
```log
[[[[[ ~\D\coauthor\coauthor ]]]]]

=> Started proxy.
=> Meteor 2.5 is available. Update this project with 'meteor update'.
=> Errors prevented startup:

   While processing files with coagmano:stylus (for target web.browser):
   fs.js:462:3: ENOENT: no such file or directory, open
   '/Users/Monish/AppData/Local/.meteor/packages/coagmano_stylus/1.1.0/plugin.compileStylusBatch.os/npm/node_modules/meteor/compileStylusBatch/node_modules/nib/lib/nib/flex.styl'
   at Object.openSync (fs.js:462:3)
   at Object.readFileSync (fs.js:364:35)
   at readFile (packages/compileStylusBatch/plugin/compile-stylus.js:169:36)
   at Array.map (<anonymous>)
   at processSourcemap (packages/compileStylusBatch/plugin/compile-stylus.js:189:52)
   at StylusCompiler.compileOneFile (packages/compileStylusBatch/plugin/compile-stylus.js:311:23)
   at getResult (packages/caching-compiler/multi-file-caching-compiler.js:113:32)
   at packages/compileStylusBatch/plugin/compile-stylus.js:61:30
   at
   C:\Users\Monish\AppData\Local\.meteor\packages\mquandalle_jade\0.4.9\plugin.compileJadeBatch.os\npm\node_modules\meteor\promise\promise\node_modules\meteor-promise\fiber_pool.js:33:40


   While processing files with coagmano:stylus (for target web.browser.legacy):
   fs.js:462:3: ENOENT: no such file or directory, open
   '/Users/Monish/AppData/Local/.meteor/packages/coagmano_stylus/1.1.0/plugin.compileStylusBatch.os/npm/node_modules/meteor/compileStylusBatch/node_modules/nib/lib/nib/flex.styl'
   at Object.openSync (fs.js:462:3)
   at Object.readFileSync (fs.js:364:35)
   at readFile (packages/compileStylusBatch/plugin/compile-stylus.js:169:36)
   at Array.map (<anonymous>)
   at processSourcemap (packages/compileStylusBatch/plugin/compile-stylus.js:189:52)
   at StylusCompiler.compileOneFile (packages/compileStylusBatch/plugin/compile-stylus.js:311:23)
   at getResult (packages/caching-compiler/multi-file-caching-compiler.js:113:32)
   at packages/compileStylusBatch/plugin/compile-stylus.js:61:30
   at
   C:\Users\Monish\AppData\Local\.meteor\packages\mquandalle_jade\0.4.9\plugin.compileJadeBatch.os\npm\node_modules\meteor\promise\promise\node_modules\meteor-promise\fiber_pool.js:33:40


   While processing files with coagmano:stylus (for target web.cordova):
   fs.js:462:3: ENOENT: no such file or directory, open
   '/Users/Monish/AppData/Local/.meteor/packages/coagmano_stylus/1.1.0/plugin.compileStylusBatch.os/npm/node_modules/meteor/compileStylusBatch/node_modules/nib/lib/nib/flex.styl'
   at Object.openSync (fs.js:462:3)
   at Object.readFileSync (fs.js:364:35)
   at readFile (packages/compileStylusBatch/plugin/compile-stylus.js:169:36)
   at Array.map (<anonymous>)
   at processSourcemap (packages/compileStylusBatch/plugin/compile-stylus.js:189:52)
   at StylusCompiler.compileOneFile (packages/compileStylusBatch/plugin/compile-stylus.js:311:23)
   at getResult (packages/caching-compiler/multi-file-caching-compiler.js:113:32)
   at packages/compileStylusBatch/plugin/compile-stylus.js:61:30
   at
   C:\Users\Monish\AppData\Local\.meteor\packages\mquandalle_jade\0.4.9\plugin.compileJadeBatch.os\npm\node_modules\meteor\promise\promise\node_modules\meteor-promise\fiber_pool.js:33:40
```

## Context
 - Device/OS: Windows (x64)
 - Coauthor server: meteor

**Comment by edemaine** [comment]
This looks like a transient network error. You should probably try deleting the Meteor package cache (`/Users/Monish/AppData/Local/.meteor/packages/`) and running Meteor again. I doubt it's a Coauthor issue.

**Comment by mynameismon** [comment]
Well, I tried that, and it doesn't fix it. It does seem like a meteor issue, though.
However, I tried to switch back to the commit tagged ``coauthor-before-react``, and that seems to work...

**Comment by edemaine** [comment]
For what it's worth, the file `AppData/Local/.meteor/packages/coagmano_stylus/1.1.0/plugin.compileStylusBatch.os/npm/node_modules/meteor/compileStylusBatch/node_modules/nib/lib/nib/flex.styl` (the one in your error message) exists for me. Does it exist for you?

**Comment by mynameismon** [comment]
That's the first thing I had checked too, and yes, it was there then.

Edit: I just checked out the latest commit and ran meteor.
```shell
> cat  /AppData/Local/.meteor/packages/coagmano_stylus/1.1.0/plugin.compileStylusBatch.os/npm/node_modules/meteor/compileStylusBatch/node_modules/nib/lib/nib/flex.styl | more
/*
 * Vendor "display: flex" support with fallback to obsolete versions.
 */

flex-version ?= box flex

//
// 1. Display values
//    - http://www.w3.org/TR/css3-flexbox/#flex-containers
//
display(type, args...)
  if flex == type || inline-flex == type
    if box in flex-version
      if flex == type
        display: -ms-flexbox args
        display: vendor-value(box args, only: moz webkit)
      else
        display: -ms-inline-flexbox args
        display: vendor-value(inline-box args, only: moz webkit)
    if flex in flex-version
-- More  --
```

**Comment by edemaine** [comment]
Are you saying it's working now? If so, yay!

**Comment by mynameismon** [comment]
I really wish! For now, I have rolled back to the version without react
Nope, I have the file, but the error still shows up after checking in the latest commit.

**Comment by edemaine** [comment, marked as answer]
Weird. I looked at the diff since version `before-react`, and the `coagmano:stylus` package was added since.  So perhaps that's the problem somehow.  You might try:

* Updating the `coagmano:stylus` dependency to `@2.0.0` (in `.meteor/packages`). This version is very slow in my experience, but it's the latest, so maybe it works?
* Opening an issue at https://github.com/Meteor-Community-Packages/meteor-stylus/issues

I'm still unable to reproduce the issue. I just tried upgrading my Meteor CLI (but not the Meteor version of Coauthor &mdash; that seems to require more work) and I can still run `meteor` fine.  I haven't tried `npm install meteor` from a clean machine though; I'm still using the Chocolately based install (so you could try that too).

**Comment by mynameismon** [comment]
Well, updating worked! Thank you for taking time out and helping 😃 

P.S. I just realised you were the same Eric Demaine from 6.006. Thank you so much for your lectures on algorithms. From the perspective of a highschool student, they were immensly helpful and wonderfully explained.

**Reply by edemaine** [reply]
Great, I'm glad it's working now! And glad you like the lectures. 🙂
